packages/stryker/src/reporters/ClearTextScoreTable.ts

```diff
@@ -0,0 +1,167 @@
+}
+
+const spaces = (n: number): string => {
+  if (n) {
+    return ` ${spaces(n - 1)}`;
+  } else {
+    return '';
+  }
+};
+
+const dots = (n: number): string => {
+  if (n) {
+    return `.${dots(n - 1)}`;
+  } else {
+    return '';
+  }
+};
+
+/**
+ * Represents a column in the clear text table
+ */
+class Column {
+  protected width: number;
+
+  constructor(public header: string, public valueFactory: TableCellValueFactory, public rows: ScoreResult) {
+    const maxContentSize = this.determineValueSize();
+    this.width = this.pad(dots(maxContentSize), maxContentSize).length;
+  }
+
+  protected determineValueSize(row: ScoreResult = this.rows, ancestorCount: number = 0): number {
+    const valueWidths = row.childResults.map(child => this.determineValueSize(child, ancestorCount + 1));
+    valueWidths.push(this.header.length);
+    valueWidths.push(this.valueFactory(row, ancestorCount).length);
+    return Math.max(...valueWidths);
+  }
+
+  /**
+   * Adds padding (spaces) to the front and end of a value
+   * @param input The string input
+   * @param targetLength The min length that has to be reached
+   */
+  protected pad(input: string, targetLength: number = this.width): string {
+    // End with space
+    if (!input.endsWith(' ')) {
+      return this.pad(input + ' ', targetLength);
+    }
+    // Start with space
+    else if (!input.startsWith(' ')) {
+      return this.pad(' ' + input, targetLength);
+    }
+    else if (input.length < targetLength) {
+      // Pad rest with spaces in front
+      return spaces(targetLength - input.length) + input;
+    } else {
+      return input;
+    }
+  }
+
+  drawLine(length = this.width): string {
+    if (length) {
+      return '-' + this.drawLine(length - 1);
+    } else {
+      return '';
+    }
+  }
+
+  drawTableCell(score: ScoreResult, ancestorCount: number) {
+    return this.color(score)(this.pad(this.valueFactory(score, ancestorCount)));
+  }
+
+  drawHeader() {
+    return this.pad(this.header);
+  }
+
+  protected color(score: ScoreResult) {
+    return (input: string) => input;
+  }
+}
+
+class MutationScoreColumn extends Column {
+  constructor(rows: ScoreResult) {
+    super('% score', row => row.mutationScore.toFixed(2), rows);
+  }
+  protected color(score: ScoreResult) {
+    if (score.mutationScore >= 80) {
+      return chalk.green;
+    } else if (score.mutationScore >= 60) {
+      return chalk.yellow;
+    } else {
+      return chalk.red;
+    }
+  }
+}
+
+class FileColumn extends Column {
+  constructor(rows: ScoreResult) {
+    super('File', (row, ancestorCount) => spaces(ancestorCount) + (ancestorCount === 0 ? FILES_ROOT_NAME : row.name), rows);
+  }
+  protected pad(input: string, targetLength: number = this.width): string {
+    if (!input.endsWith(' ')) {
+      return this.pad(`${input} `, targetLength);
+    } else if (input.length < targetLength) {
+      return input + spaces(targetLength - input.length);
```

Comment:
Can't we reduce the content of this method to just this one line and still end up with the same result?